src/java.base/share/classes/java/lang/reflect/code/interpreter/Interpreter.java

```diff
@@ -419,9 +420,9 @@ static Object exec(MethodHandles.Lookup l, OpContext oc, Op o) {
                 case SUPER -> l.in(target.parameterType(0));
             };
             MethodHandle mh = resolveToMethodHandle(il, co.invokeDescriptor(), co.invokeKind());
-
-            mh = mh.asType(target).asFixedArity();
             Object[] values = o.operands().stream().map(oc::getValue).toArray();
+            target = eraseInts(mh.type(), target, values);
```

Comment:
Can you provide an example of why this is required? I think i can guess based on changes to the lifter. Can this impact other areas too like field and array access?

I think in general we have to be careful here. The properties of lifted models are having a broader impact. Arguably, this could be a code model transformation instead.

Terminology-wise "erase" is reserved for references. IIUC correctly we are operating on values of basic primitive types and i believe converting to them non-basic types based on corresponding method parameter types.

Reply:
OK, I think I understand the direction, relaxed rules have ugly side effects. I'll change this PR the other way - lift will insert explicit int conversions to the identified places where Interpreter would clearly fail. Interpreter stay more strict and Verifier follow the Interpreter.

Reply:
Ok, we can always go with passing an option to the Interpreter if the transformation approach does not work out.